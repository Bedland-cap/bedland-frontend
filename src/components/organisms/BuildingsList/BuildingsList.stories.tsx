import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import BuildingsList from './BuildingsList';

const meta = {
  title: 'Organisms/BuildingList',
  component: BuildingsList,
  parameters: {
    backgrounds: { default: 'light' },
  },
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof BuildingsList>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
