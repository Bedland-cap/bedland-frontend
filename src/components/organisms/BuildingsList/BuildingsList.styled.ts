import styled from 'styled-components';

export const BuildingsContainer = styled.div`
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit, minmax(20rem, 1fr));
`;

export default BuildingsContainer;
