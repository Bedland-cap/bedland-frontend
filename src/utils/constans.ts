export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
