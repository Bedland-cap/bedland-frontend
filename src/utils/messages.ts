const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
