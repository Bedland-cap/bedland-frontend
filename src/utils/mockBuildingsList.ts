import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
