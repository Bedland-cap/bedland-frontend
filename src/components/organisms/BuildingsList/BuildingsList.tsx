import BuildingTile from 'components/molecules/BuildingTile/BuildingTile';
import mockBuildingsList from 'utils/mockBuildingsList';
import * as Styled from './BuildingsList.styled';

const BuildingsList = () => (
  <Styled.BuildingsContainer>
    {mockBuildingsList.map((building) => (
      <BuildingTile
        key={`building-${building.id}`}
        buildingId={building.id}
        buildingAddress={building.address}
      />
    ))}
  </Styled.BuildingsContainer>
);

export default BuildingsList;
