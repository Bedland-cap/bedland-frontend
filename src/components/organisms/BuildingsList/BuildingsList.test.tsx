import { renderWithProviders } from 'test.utils';
import BuildingsList from './BuildingsList';

describe('BuildingsList', () => {
  it('renders correctly', () => {
    const buildingsList = renderWithProviders(<BuildingsList />, {});
    expect(buildingsList).toMatchSnapshot();
  });
});
