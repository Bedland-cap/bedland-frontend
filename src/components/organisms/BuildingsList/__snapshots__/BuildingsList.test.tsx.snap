// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingsList renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSUZER cbQclI"
      >
        <div
          class="sc-bcXHqe fBLOax"
        >
          <img
            alt="Building"
            class="sc-gswNZR djSoAM"
            src=""
          />
          <div
            class="sc-dkrFOg doHqxF"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Building #
              1
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Berry Street 21
            </p>
            <div
              class="sc-hLBbgP koPdij"
            >
              <button
                class="sc-eDvSVe hLvfrA"
              >
                Open details
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe fBLOax"
        >
          <img
            alt="Building"
            class="sc-gswNZR djSoAM"
            src=""
          />
          <div
            class="sc-dkrFOg doHqxF"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Building #
              2
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Miedziana Street 3
            </p>
            <div
              class="sc-hLBbgP koPdij"
            >
              <button
                class="sc-eDvSVe hLvfrA"
              >
                Open details
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe fBLOax"
        >
          <img
            alt="Building"
            class="sc-gswNZR djSoAM"
            src=""
          />
          <div
            class="sc-dkrFOg doHqxF"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Building #
              3
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Seaport Avenue 51
            </p>
            <div
              class="sc-hLBbgP koPdij"
            >
              <button
                class="sc-eDvSVe hLvfrA"
              >
                Open details
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe fBLOax"
        >
          <img
            alt="Building"
            class="sc-gswNZR djSoAM"
            src=""
          />
          <div
            class="sc-dkrFOg doHqxF"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Building #
              4
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Desert Street 2X
            </p>
            <div
              class="sc-hLBbgP koPdij"
            >
              <button
                class="sc-eDvSVe hLvfrA"
              >
                Open details
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe fBLOax"
        >
          <img
            alt="Building"
            class="sc-gswNZR djSoAM"
            src=""
          />
          <div
            class="sc-dkrFOg doHqxF"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Building #
              5
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Lakewood Avenue 57
            </p>
            <div
              class="sc-hLBbgP koPdij"
            >
              <button
                class="sc-eDvSVe hLvfrA"
              >
                Open details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSUZER cbQclI"
    >
      <div
        class="sc-bcXHqe fBLOax"
      >
        <img
          alt="Building"
          class="sc-gswNZR djSoAM"
          src=""
        />
        <div
          class="sc-dkrFOg doHqxF"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Building #
            1
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Berry Street 21
          </p>
          <div
            class="sc-hLBbgP koPdij"
          >
            <button
              class="sc-eDvSVe hLvfrA"
            >
              Open details
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe fBLOax"
      >
        <img
          alt="Building"
          class="sc-gswNZR djSoAM"
          src=""
        />
        <div
          class="sc-dkrFOg doHqxF"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Building #
            2
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Miedziana Street 3
          </p>
          <div
            class="sc-hLBbgP koPdij"
          >
            <button
              class="sc-eDvSVe hLvfrA"
            >
              Open details
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe fBLOax"
      >
        <img
          alt="Building"
          class="sc-gswNZR djSoAM"
          src=""
        />
        <div
          class="sc-dkrFOg doHqxF"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Building #
            3
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Seaport Avenue 51
          </p>
          <div
            class="sc-hLBbgP koPdij"
          >
            <button
              class="sc-eDvSVe hLvfrA"
            >
              Open details
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe fBLOax"
      >
        <img
          alt="Building"
          class="sc-gswNZR djSoAM"
          src=""
        />
        <div
          class="sc-dkrFOg doHqxF"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Building #
            4
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Desert Street 2X
          </p>
          <div
            class="sc-hLBbgP koPdij"
          >
            <button
              class="sc-eDvSVe hLvfrA"
            >
              Open details
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe fBLOax"
      >
        <img
          alt="Building"
          class="sc-gswNZR djSoAM"
          src=""
        />
        <div
          class="sc-dkrFOg doHqxF"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.75; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Building #
            5
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Lakewood Avenue 57
          </p>
          <div
            class="sc-hLBbgP koPdij"
          >
            <button
              class="sc-eDvSVe hLvfrA"
            >
              Open details
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
